### ISSUE TITLE
questions to use the package

### ISSUE THREAD

**Issue by spree1010** [anibalsilva1/ModelOptimizationIR#1]
Hi,
I am very interested in this package, and try it in my dataset. 

Here is the output graph and the output shows the dataset is imbalanced at low and high end, is it right?
![image](https://user-images.githubusercontent.com/384790/192930892-e89b1a94-b879-4b7c-b4de-fb9693d46c70.png)

Then I try it with xgboost to predict, but the predict result seems put much importance at low end
![image](https://user-images.githubusercontent.com/384790/192931288-69570ece-2896-4394-b798-12af86246cc4.png)

would you please help to suggest how can improve

Thank you!
Kun

**Comment by anibalsilva1** [comment]
Hello @spree1010, thank you for reaching out.

Regarding your first question: _Here is the output graph and the output shows the dataset is imbalanced at the low and high end, is it right?_

The target variable seems to present some skewness towards the left side, but in the end, it is imbalanced only at the low end (where you can see the outlier). Do you have any domain knowledge for the target variable, i.e., if it should be imbalanced on both, or only for the right/left side? Ideally, domain knowledge should be used to generate $\phi(y)$.

Regarding the model, I'm assuming that you called `wf.XGBoost_SERA`, which is a wrapper to call `xgboost::xgboost`.

Did you use any hyper-parameter or did you just leave the default settings?

If possible, can you send a reproducible example?

Best regards,

Aníbal Silva

**Comment by spree1010** [comment]
Hi @anibalsilva1 
Thank you for the feedback,  sorry I am beginner for this and here is my input:

1. From domain knowledge perspective,  it is imbalanced at both low and high.  And I am wondering how to generate \phi(y). 
2. Yes, I just use your source code here   [custom_loss_functions.R](https://github.com/anibalsilva1/ModelOptimizationIR/blob/master/R/custom_loss_functions.R)
3. I  set random xgboost hyper-parameter, and the results are similar
4. Is it ok I send you the data by email?  if so would you please tell me you email address :) 

Thank you a lot!
Kun

**Comment by anibalsilva1** [comment]
Hi @spree1010,

The way you generate $\phi(y)$ seems correct, I'm assuming you're calling `phi` from IRon package?

Ideally, `phi.control` should be called before with `control.pts` as inputs to determine the control points that define `phi`.

In this absence, `phi` will be determined based on control points that are obtained in adjusted box plot statistics (median, lower and upper fence range).

When `xgboost.sera` is called, `get_extreme_type` is also called, and it infers the type of extremes the distribution faces (in this case, it is low because there is only one outlier at the lower fence). 

We are currently working on a way to determine the sigmas outside the custom loss function, so you can have more freedom in determining the type of imbalance a distribution faces.


You can send the data to anibalferreiraesilva@gmail.com.

Best regards,
Aníbal Silva
